csi/controllerserver.py

```diff
@@ -145,8 +145,13 @@ def CreateVolume(self, request, context):
 
                 # The external volume should be used as kadalu host vol
 
-                # TODO: handle the case where host-volume is full
-                # can-be-fixed-by-an-intern
+                if is_hosting_volume_free(ext_volume['name'], pvsize) is False:
+                    errmsg = "Hosting Volume '%s' is Full. Add More Storage" % ext_volume['name']
```

Comment:
Use logf to change to structured logging format.

```
errmsg = logf("Hosting Volume is full. Add more Storage", volume=ext_volume['name'])
```

Reply:
Done